File: java/sorting/bubble.java
package sorting;

import java.util.Arrays;

public class bubble {
    public static void main(String[] args) {
        int arr[] = { -1,9,-3 };
        System.out.println(Arrays.toString(bubble_sort(arr)));
    }

    static int[] bubble_sort(int arr[]) {
        boolean swapped = false;
        int n = arr.length;
        if (n <= 1)
            return arr;

        for (int i = 0; i < n; i++) {
            for (int j = 1; j < n - i; j++) {

                if (arr[j] < arr[j - 1]) {
                    // swap
                    int temp = arr[j];
                    arr[j] = arr[j - 1];
                    arr[j - 1] = temp;
                    swapped = true;
                }
            }
            if (!swapped) {
                break;
            }

        }
        return arr;
    }
}

File: arrays/sorting/bubble.cc
#include <iostream>
void bubble_sort(int arr[], int n) {
  for (int i = 0; i < n; i++) {
    bool swapped{0};
    for (int j = i + 1; j < n; j++) {
      if (arr[i] > arr[j]) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;

        swapped = 1;
      }
    }

    if (!swapped)
      break;
  }
}

int main() {
  int arr[] = {3, 2, 1};
  int n = sizeof(arr) / sizeof(arr[0]);

  bubble_sort(arr, n);

  for (int i = 0; i < n; i++) {
    std::cout << arr[i] << " ";
  }

  std::cout << std::endl;
  return 0;
}
